bridge/gitlab/iterator.go

```diff
@@ -0,0 +1,253 @@
+package gitlab
+
+import (
+	"time"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+type issueIterator struct {
+	page  int
+	index int
+	cache []*gitlab.Issue
+}
+
+type noteIterator struct {
+	page  int
+	index int
+	cache []*gitlab.Note
+}
+
+type labelEventIterator struct {
+	page  int
+	index int
+	cache []*gitlab.LabelEvent
+}
+
+type iterator struct {
+	// gitlab api v4 client
+	gc *gitlab.Client
+
+	// if since is given the iterator will query only the updated
+	// issues after this date
+	since time.Time
+
+	// project id
+	project string
+
+	// number of issues and notes to query at once
+	capacity int
+
+	// sticky error
+	err error
+
+	// issues iterator
+	issue *issueIterator
+
+	// notes iterator
+	note *noteIterator
+
+	// labelEvent iterator
+	labelEvent *labelEventIterator
+}
+
+// NewIterator create a new iterator
+func NewIterator(projectID, token string, since time.Time) *iterator {
+	return &iterator{
+		gc:       buildClient(token),
+		project:  projectID,
+		since:    since,
+		capacity: 20,
+		issue: &issueIterator{
+			index: -1,
+			page:  1,
+		},
+		note: &noteIterator{
+			index: -1,
+			page:  1,
+		},
+		labelEvent: &labelEventIterator{
+			index: -1,
+			page:  1,
+		},
+	}
+}
+
+// Error return last encountered error
+func (i *iterator) Error() error {
+	return i.err
+}
+
+func (i *iterator) getNextIssues() bool {
+	sort := "asc"
+	scope := "all"
```

Comment:
not that it's wrong, but why not use those `gitlab.String(..)` helpers ? https://github.com/xanzy/go-gitlab/blob/master/gitlab.go#L859-L863

Reply:
Didn't know they have that :+1: